### PR TITLE
[CL] Remove mapping process id to component name

### PR DESCRIPTION
**Pull request by renchao-lu**
As titled.

### CODE REVIEW

File: ChemistryLib/PhreeqcIOData/AqueousSolution.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ namespace PhreeqcIOData
 {
 struct Component
 {
-    explicit Component(std::string name_) : name(std::move(name_)) {}
+    explicit Component(std::string&& name_) : name(std::move(name_)) {}
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIOData/AqueousSolution.h:27]: minor: can be kept as value which is more general. *i.e* on can pass something constant to the ctor.

File: ChemistryLib/PhreeqcIOData/CreateSolutionComponent.cpp [hunk 2]
```diff
@@ -23,11 +23,11 @@ std::vector<Component> createSolutionComponents(
     //! \ogs_file_param{prj__chemical_system__solution__components}
     auto comp_config = config.getConfigSubtree("components");
     for (
-        auto const& component_name :
+        auto component_name :
         //! \ogs_file_param{prj__chemical_system__solution__components__component}
         comp_config.getConfigParameterList<std::string>("component"))
     {
-        components.emplace_back(component_name);
+        components.emplace_back(std::move(component_name));
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIOData/CreateSolutionComponent.cpp:30]: This indeed saves one copy.

File: ProcessLib/ComponentTransport/CreateComponentTransportProcess.h [hunk 3]
```diff
@@ -32,7 +37,8 @@ std::unique_ptr<Process> createComponentTransportProcess(
     BaseLib::ConfigTree const& config,
     std::vector<std::unique_ptr<MeshLib::Mesh>> const& meshes,
     std::string const& output_directory,
-    std::map<int, std::shared_ptr<MaterialPropertyLib::Medium>> const& media);
-
+    std::map<int, std::shared_ptr<MaterialPropertyLib::Medium>> const& media,
+    std::unique_ptr<ChemistryLib::ChemicalSolverInterface> const&
+        chemical_solver_interface);
```
Comment (endJunction) [inline ProcessLib/ComponentTransport/CreateComponentTransportProcess.h:42]: Clarification needed: With this declaration of the argument the `chemical_solver_interface` is an optional thing. Is it inteded?
If not, better to pass a `ChemistryLib::ChemicalSolverInterface const&`.

File: ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp [hunk 4]
```diff
@@ -105,7 +107,7 @@ std::unique_ptr<Process> createComponentTransportProcess(
 
     // Collect all process variables in a vector before allocation
     // pressure first, concentration then
-    auto const collected_process_variables = findProcessVariables(
+    auto collected_process_variables = findProcessVariables(
```
Comment (endJunction) [inline ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp:110]: minor: I didn't get it why it is not longer const.
I'd even make it a `const &` if possible. (Avoids a copy of a vector, though the gains are minuscule, but semantics is much better.)

File: ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp [hunk 5]
```diff
@@ -140,13 +142,49 @@ std::unique_ptr<Process> createComponentTransportProcess(
...
+            auto const components =
+                chemical_solver_interface->getComponentList();
+            // pressure
+            per_process_variable.emplace_back(collected_process_variables[0]);
+            process_variables.push_back(std::move(per_process_variable));
+            // concentration
+            assert(components.size() + 1 == collected_process_variables.size());
+            std::transform(components.begin(), components.end(),
+                           std::back_inserter(process_variables),
+                           [&per_process_variable,
```
Comment (endJunction) [inline ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp:164]: style: move the lambda one level up placing it before the transform call. Makes it readable.

File: ChemistryLib/PhreeqcIO.h [hunk 6]
```diff
@@ -77,6 +77,8 @@ class PhreeqcIO final : public ChemicalSolverInterface
 
     friend std::istream& operator>>(std::istream& in, PhreeqcIO& phreeqc_io);
 
+    std::vector<std::string> const getComponentList() const;
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIO.h:80]:
```suggestion
    std::vector<std::string> const getComponentList() const override;
```

File: ChemistryLib/PhreeqcIOData/AqueousSolution.h (hunk 1 above)
Comment (renchao-lu) [reply ChemistryLib/PhreeqcIOData/AqueousSolution.h:27]: kept the ctor as it was.

File: ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp (hunk 4 above)
Comment (renchao-lu) [reply ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp:110]: my bad. fixed.

File: ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp (hunk 5 above)
Comment (renchao-lu) [reply ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp:164]: Fixed.

File: ChemistryLib/PhreeqcIO.h (hunk 6 above)
Comment (renchao-lu) [reply ChemistryLib/PhreeqcIO.h:80]: :thumbsup: Fixed.

File: ChemistryLib/PhreeqcIO.h [hunk 7]
```diff
@@ -101,8 +101,6 @@ class PhreeqcIO final : public ChemicalSolverInterface
     std::unique_ptr<Output> const _output;
     std::unique_ptr<Dump> const _dump;
     Knobs const _knobs;
-    std::vector<std::pair<int, std::string>> const&
-        _process_id_to_component_name_map;
     double _dt = std::numeric_limits<double>::quiet_NaN();
     const int phreeqc_instance_id = 0;
```
Comment (wenqing) [inline ChemistryLib/PhreeqcIO.h:105]: Although not from this PR,  the underscore is not used as the prefix for this member name.  I will change it in my PR.

Comment (endJunction) [reply ChemistryLib/PhreeqcIO.h:105]: @wenqing  There are many more such cases. Don't mix up different topics in single PR or commit.

Comment (wenqing) [reply ChemistryLib/PhreeqcIO.h:105]: Not yet. Just want to do after the PR for name changing is merged or its current review is done.

**Pull request merged**